src/compiler/types/validate-build-package-json.ts

```diff
@@ -37,12 +53,20 @@ const validatePackageJsonOutput = async (
   await Promise.all([
     validatePackageFiles(config, compilerCtx, buildCtx, outputTarget),
     validateMain(config, compilerCtx, buildCtx, outputTarget),
-    validateModule(config, compilerCtx, buildCtx, outputTarget),
```

Comment:
Is there any particular reason we hoist the call to `validateModule` from this location to https://github.com/ionic-team/stencil/pull/3475/files#diff-00920ea8f2d029f17679521b22c40f18bbb181d9cad5e9a1339872fe58cdb6fdR35? To me, it seems to more intuitive to co-locate the validation each of the fields in the scope of a single function (`validatePackageJsonOutput`).

Reply:
Yes, the reason I moved it was that the `validatePackageJsonOutput` function is currently only called if there is a `DIST_COLLECTION` output target set on the Stencil config. Since we're adding validation specific to the `DIST_CUSTOM_ELEMENTS` output target I wanted to make sure that the code in `validateModule` would run no matter which output targets were enabled, so I moved it out of `validatePackageJsonOutput` (which, come to think of it, I should probably also rename while we're in here) and I refactored it to just look at the config, find output targets, etc.